accounting/src/delta/combine.rs

```diff
@@ -15,26 +15,46 @@
 
 use common::primitives::{signed_amount::SignedAmount, Amount};
 
-use crate::error::Error;
-
-use super::delta_data_collection::DataDelta;
+use crate::{
+    delta::delta_data_collection::{DataDelta, DataDeltaUndo, DeltaMapElement},
+    error::Error,
+};
 
+/// Combine data with an element of `DeltaDataCollection`.
+/// An element can be either a Delta or a result of delta undo.
+/// Note that the rules for these 2 cases differ.
 pub fn combine_data_with_delta<T: Clone>(
     lhs: Option<&T>,
-    rhs: Option<&DataDelta<T>>,
+    rhs: Option<&DeltaMapElement<T>>,
 ) -> Result<Option<T>, Error> {
     match (lhs, rhs) {
         (None, None) => Ok(None),
         (None, Some(d)) => match d {
-            DataDelta::Create(d) => Ok(Some(*d.clone())),
-            DataDelta::Modify(_) => Err(Error::ModifyNonexistingData),
-            DataDelta::Delete => Err(Error::RemoveNonexistingData),
+            DeltaMapElement::Delta(d) => match d {
+                DataDelta::Create(d) => Ok(Some(*d.clone())),
+                DataDelta::Modify(_) => Err(Error::ModifyNonexistingData),
+                DataDelta::Delete(_) => Err(Error::RemoveNonexistingData),
+            },
+            DeltaMapElement::DeltaUndo(u) => match u {
+                DataDeltaUndo::Create(d) => Ok(Some(*d.clone())),
+                DataDeltaUndo::Modify(_) => Err(Error::ModifyNonexistingData),
+                // This is Ok because the Create + Undo = Delete
```

Comment:
This is not Ok. I'll rework and add real tests for data + delta, considering new changes with No-op. In the mean time please see `delta_data_collection` module